### PR TITLE
bump etcd to 3.2.8

### PR DESCRIPTION
**Pull request by soltysh**
This bumps etcd and performs additional cleanups needed for our restoration to work properly.

### CODE REVIEW

**Comment by liggitt** [comment]
looks like the v2http package was renamed and is failing tests and build

**Comment by liggitt** [comment]
at least, the part we use for NewPeerHandler (https://github.com/coreos/etcd/compare/v3.2.1...v3.2.8#diff-d7a47eb75475dba263540eb7f8456e50L152)

**Comment by liggitt** [comment]
the etcd bump commit is much bigger than the [corresponding diff](https://github.com/coreos/etcd/compare/v3.2.1...v3.2.8) upstream... why is that?

File: hack/godep-restore.sh [hunk 1]
```diff
@@ -56,8 +56,7 @@ preload-remote "github.com/openshift" "origin" "github.com/openshift" "origin" #
 preload-remote "k8s.io" "kubernetes" "github.com/openshift" "kubernetes"
 preload-remote "github.com/docker" "distribution" "github.com/openshift" "docker-distribution"
 preload-remote "github.com/skynetservices" "skydns" "github.com/openshift" "skynetservices-skydns"
-preload-remote "github.com/coreos" "etcd" "github.com/openshift" "coreos-etcd"
```
Comment (liggitt) [inline hack/godep-restore.sh]: @smarterclayton why did we have this?
@soltysh do we no longer have any upstream commits?

Comment (soltysh) [reply hack/godep-restore.sh]: I think we had those some time ago, but I've double checked the last bump and it was using official version (no carry commits). That's why I've decided to remove it from here. We'll bring it back when needed.

**Comment by soltysh** [comment]
> the etcd bump commit is much bigger than the corresponding diff upstream... why is that?

I'll double check, but I'm guessing we were missing some tests, or alike.

**Comment by soltysh** [comment]
> the etcd bump commit is much bigger than the corresponding diff upstream... why is that?

I've double checked and the reasons the bump is bigger than the difference are as follows:

1. Updated tests, a lot of them. I'm guessing previous bump was done without `-t` flag, which is enabled by default when invoking `hack/godep-save.sh` which I've used here.
2. Removed packages (eg. `github.com/coreos/etcd/clientv3/namespace` and `github.com/coreos/etcd/clientv3/naming`), looks like we don't need those anymore. 
3. Generated files (eg. `github.com/coreos/etcd/etcdserver/etcdserverpb/rpc.pb.gw.go`) removed in this bump. 

The tests are green on it, but I'll leave that up to you to decide @smarterclayton @liggitt @deads2k

**Comment by soltysh** [comment]
@liggitt I've added that no-op bump commit for 3.2.1. With that in the actual 3.2.8 bump is a subset of https://github.com/coreos/etcd/compare/v3.2.1...v3.2.8
I wonder though, why we vendor integration tests for etcd, any particular reason, @smarterclayton ?

**Comment by smarterclayton** [comment]
We don't need them.


On Oct 11, 2017, at 1:12 PM, Maciej Szulik <notifications@github.com> wrote:

@liggitt <https://github.com/liggitt> I've added that no-op bump commit for
3.2.1. With that in the actual 3.2.8 bump is a subset of coreos/etcd@
v3.2.1...v3.2.8 <https://github.com/coreos/etcd/compare/v3.2.1...v3.2.8>
I wonder though, why we vendor integration tests for etcd, any particular
reason, @smarterclayton <https://github.com/smarterclayton> ?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/16730#issuecomment-335881056>,
or mute
the thread

**Comment by soltysh** [comment]
> We don't need them.

I'll do one another sweep through Godeps and I'll try removing them, in that case.

**Comment by liggitt** [comment]
> We don't need them.
> 
> I'll do one another sweep through Godeps and I'll try removing them, in that case.

was the content of Godeps.json in this PR not the result of `hack/save-godeps.sh`? If so, I wouldn't worry about it for this PR

**Comment by soltysh** [comment]
> was the content of Godeps.json in this PR the result of hack/save-godeps.sh? If so, I wouldn't worry about it for this PR

Yes, that's why I think we somehow rely on them. But maybe we don't any more, I'm not sure how good godep is with figuring out those deps.

**Pull request merged**